### PR TITLE
feat(replays): Update the Replay Index page columns and Details page header to match

### LINKED ISSUE #42784: Replay: Display browser information
### Problem Statement

Browser detail such as Name and Version can be useful to help narrow down and reproduce of issues.

Related: Customers have requested that the browser window size and resolution to be available.
* https://github.com/getsentry/sentry/issues/42774

### Solution Brainstorm

Similar to how we present on error events:
image

### Designs
https://www.figma.com/file/fOXcP5EaaTnerWqrydnyCP/Exploration%3A-Index-v3?node-id=190%3A38288&t=8LQE1kEXtxEeziw9-1

### LINKED ISSUE #44759: Consistent UI for primary meta data
## Problem
- With [#44345](https://github.com/getsentry/sentry/issues/44345) replacing all event tables with a hashed replay ID to help users find a specific replay, we need to show this replay ID in the details and index page now
- In the replay details page, we need to relocate the user identifier("[User Scrubbed]", IP address, email, or username) to accommodate the replay ID position in the header
- In the replay index, we have to show "[User Scrubbed]" when IP address are scrubbed from a replay
- In the replay index and details, we have to allocate more space for OS and Browser information

## Related Tickets
- https://github.com/getsentry/sentry/issues/44345 ← ⚠ This has to be completed before starting this ticket
- https://github.com/getsentry/sentry/issues/42784

## Designs
https://www.figma.com/file/fOXcP5EaaTnerWqrydnyCP/Exploration%3A-Index-v3?node-id=190%3A38289&t=Ps3r1Eoz1dHP0zF5-1

### Index
Image
Image

### PR DESCRIPTION
**Pull request by ryan953**
New css in the app to match the [figma](https://www.figma.com/file/fOXcP5EaaTnerWqrydnyCP/Specs%3A-Index-v3?node-id=190-38289&t=swXsyTNh1ysgRSI6-0):

### Table rows updated:
SCR-20230316-j17

### Details page header updated:
| With error count | No errors | Loading
| --- | --- | --- |
| SCR-20230316-j18  | SCR-20230316-kn5 | SCR-20230316-kmu |

Here's the diff of the header loading state and the header with info:
SCR-20230316-kni

Fixes #42784
Fixes #44759

### CODE REVIEW

**Comment by billyvg** [comment]
image
So many numbers, I wonder how important the version numbers are? What if we moved the versionto a tooltip?

File: static/app/components/replays/contextIcon.tsx [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+const LazyContextIcon = lazy(
+  () => import('sentry/components/events/contextSummary/contextIcon')
+);
+
+const ContextIcon = styled(({className, name, version}: Props) => {
+  const icon = generateIconName(name, version ?? undefined);
+  return (
+    <div className={className}>
+      <Tooltip title={name}>
+        <Suspense fallback={<LoadingMask />}>
```
Comment (billyvg) [inline static/app/components/replays/contextIcon.tsx:24]: Will the mask have a proper size?

File: static/app/views/replays/detail/page.tsx [hunk 2]
```diff
@@ -33,12 +34,28 @@ function Page({children, crumbs, orgSlug, replayRecord}: Props) {
...
-        <CrumbWalker replayRecord={replayRecord} crumbs={crumbs} />
+        <UserBadge
+          avatarSize={32}
+          displayName={<DisplayName>{replayRecord.user.display_name || ''}</DisplayName>}
+          user={{
+            username: replayRecord.user.display_name || '',
+            email: replayRecord.user.email || '',
+            id: replayRecord.user.id || '',
+            ip_address: replayRecord.user.ip || '',
+            name: replayRecord.user.username || '',
```
Comment (billyvg) [inline static/app/views/replays/detail/page.tsx:50]: Has this been fixed? I remember a support ticket where user was trying to search via username, but it was being aliased to `name` on the BE.

**Comment by Jesse-Box** [comment]
@ryan953 looking good mate, just a few things I can see which are different from my designs
1. The user avatars are a little big. It should be 24px in size
2. The project badge sitting to the left of the replay ID is a tad too big. Looks to be about 16px, when it should be 12px in size.
3. The font-size of the error count (Details Page) is smaller than, for example, the start time
4. The user identifier string should be bold/medium font-weight.

File: static/app/components/replays/contextIcon.tsx (hunk 1 above)
Comment (ryan953) [reply static/app/components/replays/contextIcon.tsx:24]: yes somehow!

i'll get a screen shot of the loading state

File: static/app/views/replays/detail/page.tsx (hunk 2 above)
Comment (ryan953) [reply static/app/views/replays/detail/page.tsx:50]: i'll check what the backend has and get that fixed. 

We should have `user.username` on the ReplayRecord, per the docs: https://docs.sentry.io/product/sentry-basics/search/searchable-properties/

This UserBadge component accepts a few variations, i think we'll have to keep this mapping the way it is.

**Comment by ryan953** [comment]
@Jesse-Box  following up:

1. The user avatars are a little big. It should be 24px in size
   - updated the avatar on the list page to be 24px
   - avatar on the details page is still 32px
3. The project badge sitting to the left of the replay ID is a tad too big. Looks to be about 16px, when it should be 12px in size.
  - updated badge on list page to be 12px
  - badge on details page (see `Session Replay > $BADGE $REPLAY_ID (NEW)`) is still 16px
5. The font-size of the error count (Details Page) is smaller than, for example, the start time
  - fixed, was missing the `<KeyMetricData>` wrapper
7. The user identifier string should be bold/medium font-weight.
  - fixed, using `<Layout.Title>` instead of a new styled component (I left a todo for this, it's to-done now)

File: static/app/views/issueDetails/groupReplays/groupReplays.tsx [hunk 3]
```diff
@@ -19,9 +17,16 @@ type Props = {
...
+  ReplayColumns.os,
+  ReplayColumns.browser,
+  ReplayColumns.duration,
+  ReplayColumns.countErrors,
+  ReplayColumns.activity,
+];
+
 function GroupReplays({group}: Props) {
-  const theme = useTheme();
-  const hasRoomForColumns = useMedia(`(min-width: ${theme.breakpoints.small})`);
```
Comment (billyvg) [inline static/app/views/issueDetails/groupReplays/groupReplays.tsx:24]: How does this look at small breakpoints now?

File: static/app/views/replays/replayTable/tableCell.tsx [hunk 4]
```diff
@@ -30,43 +34,64 @@ export function SessionCell({
...
+        displayName={
+          <MainLink to={replayDetails}>{replay.user.display_name || ''}</MainLink>
+        }
+        user={{
+          username: replay.user.display_name || '',
+          email: replay.user.email || '',
+          id: replay.user.id || '',
+          ip_address: replay.user.ip || '',
+          name: replay.user.username || '',
+        }}
```
Comment (billyvg) [inline static/app/views/replays/replayTable/tableCell.tsx:58]: I think we do this in at least one other place... might be worth refactoring into a util fn.

**Review by billyvg — APPROVED** [review]

File: static/app/views/issueDetails/groupReplays/groupReplays.tsx (hunk 3 above)
Comment (ryan953) [reply static/app/views/issueDetails/groupReplays/groupReplays.tsx:24]: I'll take some screen shots!

**Comment by billyvg** [comment]
Some additional design grumblings... The avatars here introduce a lot of vertical whitespace 
image


They also encompass an additional line of information now as it's centered between user, url walker, and now project + timestamp.


The layout here is a bit broken too:
image
I would expect it to take up the full width and display the entire URL instead of truncating.

**Comment by billyvg** [comment]
Oh, this is a blocker: the project is wrong, looks like it's showing a short id for the replay? Or is that intended?

**Comment by billyvg** [comment]
Ah noticed a case I missed re: url walker..., if it's exactly 3 items the middle will not shrink

**Comment by ryan953** [comment]
> Oh, this is a blocker: the project is wrong, looks like it's showing a short id for the replay? Or is that intended? (If it's intended then I find that very confusing).

Yes. We've got the Project Icon + Replay-shortID
I think the project icon is not super important to have, and the replay_id is something new that we're showing... it's in the tables around the app already (discover, issue>replays, etc) so we need it in this table, in the details page header someplace. I think we'll put it in and see how that goes. It could be something to lift above the username/ip address too

**Pull request merged**